Firestore/Source/Remote/FSTRemoteStore.mm

```diff
@@ -282,7 +282,16 @@ - (void)recordPendingRequestForTargetID:(FSTBoxedTargetID *)targetID {
  * active watch targets.
  */
 - (BOOL)shouldStartWatchStream {
-  return [self isNetworkEnabled] && ![self.watchStream isStarted] && self.listenTargets.count > 0;
+  return [self shouldStartWatchStreamModuloListenerCheck] && self.listenTargets.count > 0;
+}
+
+/**
+ * Similar to shouldStartWatchStream except this one does not check whether there are active watch
+ * target or not. For certain cases, even there is no listener, we may still want to keep the watch
+ * stream live, see http://github.com/firebase/firebase-ios-sdk/issues/1165.
+ */
+- (BOOL)shouldStartWatchStreamModuloListenerCheck {
```

Comment:
Maybe just `shouldStartWatchStreamAfterError`?

Also you'll need to add some calls to this.